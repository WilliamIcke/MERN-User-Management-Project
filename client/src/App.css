html, body, #root, main {
  display: flex;
  text-align: center;
  flex-direction: column;
  height: 100%;
  width: 100%;
  background-color: #A5D0A8;
}

.Card-Header {
  background-color: #246A73;
  color: white;
}

.Card-Body {
  display: flex;
  text-align: center;
  flex-direction: column;
  width: 100%;
  background-color: #A5D0A8;
  color: black;
  padding-top: 10px;
}

/* User record list */
.userRecordListContainer {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  align-content: center;
  gap: 10px;
}

.userRecordCard {
  display: flex;
  flex-direction: column;
  padding: 5px;
  border: 3px solid #246A73;
  border-radius: 12px;
  max-width: 90%;
}

.userRecordCard h1 {
  margin-bottom: 2px;
}

.userRecordCard dl {
  margin-top: 2px;
}

.userRecordCard dt {
  font-weight: bold;
  text-decoration: underline;
}
.userRecordCard dd {
  margin: 0;
  padding: 0 0 0.5em 0;
}

p, li, dt, dd {
  font-size: 20px;
}

/* Form */
input, select {
  width: 75%;
  min-width: 100px;
  max-width: 500px;
  padding: 6px;
  border: 1px solid #ccc;
  border-radius: 4px;
  resize: vertical;
  font-size: 20px;
}

.activeCheckbox {
  height: 25px;
  width: 25px;
}

label {
  padding: 12px 12px 12px 0;
  display: inline-block;
  font-size: 20px;
}

.container {
  border-radius: 5px;
  background-color: #f2f2f2;
  padding: 20px;
}

.row:after {
  content: "";
  display: table;
  clear: both;
}

/** Form Edit Modal **/
.editModalContent {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  vertical-align: center;
  justify-content: center;
}

.EditForm .form-container {
  height: 100%;
}

.EditForm button.UserSubmit {
  min-width: 100px;
}

.editModalContent button.closeEditModal, .editModalContent button.userSubmit, .editUserButton {
  border: none;
  padding: 24px 40px;
  min-width: 100px;
  margin-left: auto;
  margin-right: auto;
  margin-top: 20px;
  margin-bottom: 20px;
  font-size: 20px;
}

.validationMessage {
  min-height: 30px;
}
