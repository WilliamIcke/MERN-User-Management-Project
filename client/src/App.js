import React from "react";
import logo from './logo.svg';
import UserRecord from './components/UserRecord'
import './App.css';
import { getUserRecords, getMembershipRecords, updateUserRecord } from './API'

const App = () => {
  const [userRecords, setUserRecords] = React.useState([]);
  const [membershipRecords, setMembershipRecords] = React.useState([]);

   React.useEffect(() => {
    fetchUserRecords();
    fetchMembershipRecords();
   }, [])

    /* Definitions of calls that use the API */
    const fetchUserRecords = () => {
        getUserRecords()
            .then(({ data: { userRecords } }) => setUserRecords(userRecords))
            .catch((err) => console.log(err))
    }

    const fetchMembershipRecords = () => {
        getMembershipRecords()
            .then(({ data: { membershipRecords } }) => setMembershipRecords(membershipRecords))
            .catch((err) => console.log(err))
    }

    const handleUpdateUserRecord = (userRecord) => {
        updateUserRecord(userRecord)
            .then(({ status, data }) => {
                if (status !== 200) {
                    throw new Error('Error! User not updated');
                }
                setUserRecords(data.userRecords);
            })
            .catch((err) => console.log(err));
    }

  return (
      <main className='App'>
        <div className='Card-Header'>
            <h1>Technical test - William Icke</h1>
            <h2>User management system</h2>
        </div>
        <div className='Card-Body userRecordListContainer'>
            {userRecords.map((userRecord) => (
                <UserRecord
                    key={userRecord._id}
                    userRecord={userRecord}
                    membershipRecords={membershipRecords}
                    updateUserRecord={handleUpdateUserRecord}
                />
            ))}
        </div>
      </main>
  )
}

export default App;
