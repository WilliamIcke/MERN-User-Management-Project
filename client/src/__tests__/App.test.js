import React from 'react';
import { shallow, mount, render } from 'enzyme';
import App from '../App';

const testData = {
  userRecord: {
      _id: 1,
      firstname: 'Test',
      surname: 'Tester',
      dateOfBirth: '12/12/1990',
      membership: 1,
      membershipStart: '12/07/2018',
      active: true,
  },
  membershipRecords: [
      {
      _id: 1,
      membershipName: 'No Membership',
      cost: '0',
      active: true
      },
      {
      _id: 2,
      membershipName: 'Priority Membership',
      cost: '250',
      active: true
      },
      {
      _id: 3,
      membershipName: 'VIP Membership',
      cost: '500',
      active: true
      },
      {
      _id: 4,
      membershipName: 'Tier 1 Membership',
      cost: '120',
      active: true
      },
      {
      _id: 5,
      membershipName: 'Basic Membership',
      cost: '50',
      active: true
      }
  ]};

describe("Component shallow snapshot tests (debug)", () => {
  test('Check App component, only', () => {
    const component =  shallow(<App debug/>);
    expect(component).toMatchSnapshot();
  });
});

describe("Component render snapshot test (debug)", () => {
  test('Check render app component', () => {
    const component =  render(<App debug/>);
    expect(component).toMatchSnapshot();
  });
});
