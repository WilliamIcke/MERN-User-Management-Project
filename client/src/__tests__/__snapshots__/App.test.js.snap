// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render snapshot test (debug) Check render app component 1`] = `
<main
  class="App"
>
  <div
    class="Card-Header"
  >
    <h1>
      Technical test - William Icke
    </h1>
    <h2>
      User management system
    </h2>
    <div
      class="statisticsListCard"
    >
      <h3>
        Membership statistics
      </h3>
    </div>
  </div>
  <div
    class="Card-Body userRecordListContainer"
  />
</main>
`;

exports[`Component shallow snapshot tests (debug) Check App component, only 1`] = `
<main
  className="App"
>
  <div
    className="Card-Header"
  >
    <h1>
      Technical test - William Icke
    </h1>
    <h2>
      User management system
    </h2>
    <div
      className="statisticsListCard"
    >
      <h3>
        Membership statistics
      </h3>
    </div>
  </div>
  <div
    className="Card-Body userRecordListContainer"
  />
</main>
`;
